src/build123d/geometry.py

```diff
@@ -1376,149 +1421,80 @@ class Location:
-                ):
-                    angle = radians(args[1])
-                    quaternion = gp_Quaternion()
-                    quaternion.SetEulerAngles(
-                        self._rot_order_dict[ordering], 0, 0, angle
-                    )
-                    transform.SetRotation(quaternion)
-
-                # set translation part after setting rotation (if exists)
-                transform.SetTranslationPart(Vector(args[0]).wrapped)
-            else:
-                translation, origin = args
-                coordinate_system = gp_Ax3(
-                    Vector(origin).to_pnt(),
-                    translation.z_dir.to_dir(),
-                    translation.x_dir.to_dir(),
-                )
-                transform.SetTransformation(coordinate_system)
-                transform.Invert()
-        elif len(args) == 3:
-            if (
-                isinstance(args[0], (Vector, Iterable))
-                and isinstance(args[1], (Vector, Iterable))
-                and isinstance(args[2], (int, float))
-            ):
-                translation, axis, angle = args
-                transform.SetRotation(
-                    gp_Ax1(Vector().to_pnt(), Vector(axis).to_dir()), angle * pi / 180.0
-                )
-            elif (
-                isinstance(args[0], (Vector, Iterable))
-                and isinstance(args[1], (Vector, Iterable))
-                and isinstance(args[2], (Extrinsic, Intrinsic))
-            ):
-                translation = args[0]
-                rotation = [radians(a) for a in args[1]]
-                ordering = args[2]
-                quaternion = gp_Quaternion()
-                quaternion.SetEulerAngles(self._rot_order_dict[ordering], *rotation)
-                transform.SetRotation(quaternion)
-            else:
-                raise TypeError("Unsupported argument types for Location")
+        if plane:
+            coordinate_system = gp_Ax3(
+                plane._origin.to_pnt(),
+                plane.z_dir.to_dir(),
+                plane.x_dir.to_dir(),
+            )
+            transform.SetTransformation(coordinate_system)
+            transform.Invert()
+        if location:
+            self.wrapped = location.wrapped
+        if top_loc:
+            self.wrapped = top_loc
+        if gp_trsf:
+            transform = gp_trsf
+        if ordering:
+            rot_order = self._rot_order_dict[ordering]
+        if orientation:
+            o_radians = [radians(a) for a in orientation]
+            quaternion = gp_Quaternion()
+            quaternion.SetEulerAngles(rot_order, *o_radians)
+            transform.SetRotation(quaternion)
+        if position:
+            # translation part must be done after SetRotation
+            transform.SetTranslationPart(Vector(position).wrapped)
 
-            transform.SetTranslationPart(Vector(translation).wrapped)
-        self.wrapped = TopLoc_Location(transform)
+        if hasattr(self, "wrapped"):
```

Comment:
I don't think having a return in the constructor is recommend. How about just using a `_wrapped` or similar variable in the processing and check `if _wrapped not None:` assign `self.wrapped = _wrapped` else use the `transform`?

Reply:
Yeah great point, what I did here was based on what init did earlier. I agree with your idea.